.github/workflows/npm-publish.yml

```diff
@@ -0,0 +1,49 @@
+# This workflow will run tests using node and then publish a
+# package to GitHub Packages when a release is created
+# For more information see:
+# https://help.github.com/actions/language-and-framework-guides/publishing-nodejs-packages
+
+name: Node.js Package
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - run: npm ci
+      - run: npm test
```

Comment:
Does this belong in a separate workflow? - One that runs on pushes?

Reply:
No idea. I'm still working out exactly what's going on here and what the process will be. I think the answer is "no", it belongs here... and the fuller answer is: "no, but another workflow probably needs to be created which includes this step, so that it can be run from PRs?"... does that sound right? That way this one runs after `semantic-release` detects a push to `master` and creates a release, so that the release artefacts are published? Then the other workflow checks (or helps check) that the PR is ok to merge.

Reply:
Well if you are going to have the tests run in a separate PR as well then I guess the question is, why do you need to run the tests again here? (They _should_ pass.)

Also if you wanted it for safety, shouldn't it just be a 'step' in the publish job. That way, if the tests fail you know nothing will be published.

Reply:
I think the PR works in the same way as drone's strange process where it cobbles the code together... then once the merge to `master` happens it does a sort of sense check. :man_shrugging:

Reply:
Well I don't share your optimism but there's not much harm in merging and finding out.